src/main/java/ch/njol/skript/expressions/ExprAllCommands.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.expressions;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.command.Commands;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+
+@Name("All commands")
+@Description("Returns all registered commands or all script commands.")
+@Examples({"send \"Number of all commands: %size of all commands%\"",
+	"send \"Number of all script commands: %size of all script commands%\""})
+@Since("INSERT VERSION")
+public class ExprAllCommands extends SimpleExpression<String> {
+	
+	static {
+		Skript.registerExpression(ExprAllCommands.class, String.class, ExpressionType.SIMPLE, "[all] [registered] (|1¦script) commands");
```

Comment:
You didn't make `scripts` optional

Reply:
I did

Reply:
No, what you did there is use a parser mark, but that doesn't make it optional. If you want it to be optional, you explicitly have to put the optional braces ([]). Like so:
```
[all] [the] [registered] [(1¦script)] commands
```

Reply:
I don't think you noticed but I have (**|**1¦script)

Reply:
see the (| ?

Reply:
that's still the same thing. And as far I remember, there's no code that makes `(|<number>¦` optional

Reply:
Ah, I see what you mean now. Still, it's better to use the optional brackets, users won't understand it is optional otherwise and it is not very clear from the first glance.